### PR TITLE
Allow kwargs for BaseException.__init__

### PR DESCRIPTION
**Pull request by gladhorn**
Exceptions take keyword args, let mypy accept those.

### CODE REVIEW

**Comment by gladhorn** [comment]
I'm at a loss for the failed test, can someone explain what it does? Is this something I broke?

**Comment by srittau** [comment]
You need to copy `stdlib/2and3/builtins.pyi` to `stdlib/2/__builtin__.pyi`. These two files need to remain synchronized.

**Comment by gladhorn** [comment]
Thanks :) I would have expected the build system to do such copying.

File: stdlib/2/__builtin__.pyi [hunk 1]
```diff
@@ -1438,7 +1438,7 @@ class BaseException(object):
         __cause__: Optional[BaseException]
         __context__: Optional[BaseException]
         __traceback__: Optional[TracebackType]
-    def __init__(self, *args: object) -> None: ...
+    def __init__(self, *args: object, **kwargs) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/2/__builtin__.pyi]: Can you add `: object` as the type? We like to keep all arguments typed.

Comment (gladhorn) [reply stdlib/2/__builtin__.pyi]: Makes sense, thanks.

**Comment by kashano** [comment]
I recently updated mypy to the latest version and started facing an issue: Too many arguments for "__init__" of "BaseException".

Since it used to work, I tried to figure out what went wrong and I found these two pull requests:

https://github.com/python/typeshed/pull/1704
https://github.com/python/typeshed/pull/2348

One to add kwargs, the second one to remove them. What's the correct behavior? Thanks!

**Comment by JelleZijlstra** [comment]
And I merged both of them, that's embarrassing. :)

I think removing them is right: BaseException really does not take kwargs
```
>>> BaseException(x=3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: exceptions.BaseException does not take keyword arguments
```

**Comment by srittau** [comment]
I agree with @JelleZijlstra. Can you give examples of code that is wrongly flagged by mypy without this patch? The example from python/mypy#4183 (referenced from #1704) is actually an example of why keyword arguments should not be allowed:

```python
class ValidationError(Exception):
    def __init__(self, message: str, log_level: int=logging.ERROR, *args, **kwargs) -> None:
        ...
        super().__init__(message, log_level, *args, **kwargs)
```

Calling `ValidationError` with keyword arguments will fail at runtime:

```python
ValidationError("", foo=123)  # TypeError: ValidationError does not take keyword arguments
```

**Comment by kashano** [comment]
I didn't knew BaseException didn't accept kwargs, mypy wasn't showing any errors until I updated it yesterday. If no kwargs is the correct behavior then I'll update my code, I just wanted to point out the conflict between versions, and if you merge this one then the next version will work differently again.

**Comment by JelleZijlstra** [comment]
Well, if your code is showing errors for code that's working correctly, there might still be a problem somewhere. Can you share the problematic code, and did you confirm that it does work correctly at runtime?

**Comment by gladhorn** [comment]
At least in my case, I mis-interpreted the class and added the kwargs, which was clearly wrong, it hid a mistake in one exception definition we had, so removing them again is certainly correct.

**Pull request closed without merging**